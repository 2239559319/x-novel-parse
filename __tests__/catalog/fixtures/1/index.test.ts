
const { join } = require("node:path");
const fs = require("fs");
const nodefetch = require("node-fetch");
const { parseCatalog } = require("../../../../dist/cjs/index-es6");

describe("load catalog config", () => {
  beforeAll(() => {
    global.fetch = nodefetch;
  });

  const filepath = join(__dirname, "./index.html");
  const html = fs.readFileSync(filepath, "utf-8");

  it("test catalog parse https://www.trongyi.vip/109_109270/", async () => {
    document.documentElement.innerHTML = html;
    Object.defineProperty(window, "location", {
      writable: true,
      value: new URL("https://www.trongyi.vip/109_109270/"),
    });

    const res = await parseCatalog(document);
    expect(res).toMatchSnapshot();
  });
});
