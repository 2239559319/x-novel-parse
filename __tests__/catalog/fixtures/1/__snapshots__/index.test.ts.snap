// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load catalog config test catalog parse https://www.trongyi.vip/109_109270/ 1`] = `
[
  {
    "name": "第 1 章",
    "url": "https://www.trongyi.vip/109_109270/1.html",
  },
  {
    "name": "第 2 章",
    "url": "https://www.trongyi.vip/109_109270/2.html",
  },
  {
    "name": "第 3 章",
    "url": "https://www.trongyi.vip/109_109270/3.html",
  },
  {
    "name": "第 4 章",
    "url": "https://www.trongyi.vip/109_109270/4.html",
  },
  {
    "name": "第 5 章",
    "url": "https://www.trongyi.vip/109_109270/5.html",
  },
  {
    "name": "第 6 章",
    "url": "https://www.trongyi.vip/109_109270/6.html",
  },
  {
    "name": "第 7 章",
    "url": "https://www.trongyi.vip/109_109270/7.html",
  },
  {
    "name": "第 8 章",
    "url": "https://www.trongyi.vip/109_109270/8.html",
  },
  {
    "name": "第 9 章",
    "url": "https://www.trongyi.vip/109_109270/9.html",
  },
  {
    "name": "第 10 章",
    "url": "https://www.trongyi.vip/109_109270/10.html",
  },
  {
    "name": "第 11 章",
    "url": "https://www.trongyi.vip/109_109270/11.html",
  },
  {
    "name": "第 12 章",
    "url": "https://www.trongyi.vip/109_109270/12.html",
  },
  {
    "name": "第 13 章",
    "url": "https://www.trongyi.vip/109_109270/13.html",
  },
  {
    "name": "第 14 章",
    "url": "https://www.trongyi.vip/109_109270/14.html",
  },
  {
    "name": "第 15 章",
    "url": "https://www.trongyi.vip/109_109270/15.html",
  },
  {
    "name": "第 16 章",
    "url": "https://www.trongyi.vip/109_109270/16.html",
  },
  {
    "name": "第 17 章",
    "url": "https://www.trongyi.vip/109_109270/17.html",
  },
  {
    "name": "第 18 章",
    "url": "https://www.trongyi.vip/109_109270/18.html",
  },
  {
    "name": "第 19 章",
    "url": "https://www.trongyi.vip/109_109270/19.html",
  },
  {
    "name": "第 20 章",
    "url": "https://www.trongyi.vip/109_109270/20.html",
  },
  {
    "name": "第 21 章",
    "url": "https://www.trongyi.vip/109_109270/21.html",
  },
  {
    "name": "第 22 章",
    "url": "https://www.trongyi.vip/109_109270/22.html",
  },
  {
    "name": "第 23 章",
    "url": "https://www.trongyi.vip/109_109270/23.html",
  },
  {
    "name": "第 24 章",
    "url": "https://www.trongyi.vip/109_109270/24.html",
  },
  {
    "name": "第 24 章",
    "url": "https://www.trongyi.vip/109_109270/25.html",
  },
  {
    "name": "第 26 章",
    "url": "https://www.trongyi.vip/109_109270/26.html",
  },
  {
    "name": "第 27 章",
    "url": "https://www.trongyi.vip/109_109270/27.html",
  },
  {
    "name": "第 28 章",
    "url": "https://www.trongyi.vip/109_109270/28.html",
  },
  {
    "name": "第 29 章",
    "url": "https://www.trongyi.vip/109_109270/29.html",
  },
  {
    "name": "第 30 章",
    "url": "https://www.trongyi.vip/109_109270/30.html",
  },
  {
    "name": "第 31 章",
    "url": "https://www.trongyi.vip/109_109270/31.html",
  },
  {
    "name": "第 32 章",
    "url": "https://www.trongyi.vip/109_109270/32.html",
  },
  {
    "name": "第 33 章",
    "url": "https://www.trongyi.vip/109_109270/33.html",
  },
  {
    "name": "第 34 章",
    "url": "https://www.trongyi.vip/109_109270/34.html",
  },
  {
    "name": "第 35 章",
    "url": "https://www.trongyi.vip/109_109270/35.html",
  },
  {
    "name": "第 36 章",
    "url": "https://www.trongyi.vip/109_109270/36.html",
  },
  {
    "name": "第 37 章",
    "url": "https://www.trongyi.vip/109_109270/37.html",
  },
  {
    "name": "第 38 章",
    "url": "https://www.trongyi.vip/109_109270/38.html",
  },
  {
    "name": "第 39 章",
    "url": "https://www.trongyi.vip/109_109270/39.html",
  },
  {
    "name": "第 40 章",
    "url": "https://www.trongyi.vip/109_109270/40.html",
  },
  {
    "name": "第 41 章",
    "url": "https://www.trongyi.vip/109_109270/41.html",
  },
  {
    "name": "第 42 章",
    "url": "https://www.trongyi.vip/109_109270/42.html",
  },
  {
    "name": "第 43 章",
    "url": "https://www.trongyi.vip/109_109270/43.html",
  },
  {
    "name": "第 44 章",
    "url": "https://www.trongyi.vip/109_109270/44.html",
  },
  {
    "name": "第 45 章",
    "url": "https://www.trongyi.vip/109_109270/45.html",
  },
  {
    "name": "第 46 章",
    "url": "https://www.trongyi.vip/109_109270/46.html",
  },
  {
    "name": "第 47 章",
    "url": "https://www.trongyi.vip/109_109270/47.html",
  },
  {
    "name": "第 48 章",
    "url": "https://www.trongyi.vip/109_109270/48.html",
  },
  {
    "name": "第 49 章",
    "url": "https://www.trongyi.vip/109_109270/49.html",
  },
  {
    "name": "第 50 章",
    "url": "https://www.trongyi.vip/109_109270/50.html",
  },
  {
    "name": "第 51 章",
    "url": "https://www.trongyi.vip/109_109270/51.html",
  },
  {
    "name": "第 52 章",
    "url": "https://www.trongyi.vip/109_109270/52.html",
  },
  {
    "name": "第 53 章",
    "url": "https://www.trongyi.vip/109_109270/53.html",
  },
  {
    "name": "第 54 章",
    "url": "https://www.trongyi.vip/109_109270/54.html",
  },
  {
    "name": "第 55 章",
    "url": "https://www.trongyi.vip/109_109270/55.html",
  },
  {
    "name": "第 56 章",
    "url": "https://www.trongyi.vip/109_109270/56.html",
  },
  {
    "name": "第 57 章",
    "url": "https://www.trongyi.vip/109_109270/57.html",
  },
  {
    "name": "第 58 章",
    "url": "https://www.trongyi.vip/109_109270/58.html",
  },
  {
    "name": "第 59 章",
    "url": "https://www.trongyi.vip/109_109270/59.html",
  },
  {
    "name": "第 60 章",
    "url": "https://www.trongyi.vip/109_109270/60.html",
  },
  {
    "name": "第 61 章",
    "url": "https://www.trongyi.vip/109_109270/61.html",
  },
  {
    "name": "第 62 章",
    "url": "https://www.trongyi.vip/109_109270/62.html",
  },
  {
    "name": "第 63 章",
    "url": "https://www.trongyi.vip/109_109270/63.html",
  },
  {
    "name": "第 64 章",
    "url": "https://www.trongyi.vip/109_109270/64.html",
  },
  {
    "name": "第 65 章",
    "url": "https://www.trongyi.vip/109_109270/65.html",
  },
  {
    "name": "第 66 章",
    "url": "https://www.trongyi.vip/109_109270/66.html",
  },
  {
    "name": "第 67 章",
    "url": "https://www.trongyi.vip/109_109270/67.html",
  },
  {
    "name": "第 68 章",
    "url": "https://www.trongyi.vip/109_109270/68.html",
  },
  {
    "name": "第 69 章",
    "url": "https://www.trongyi.vip/109_109270/69.html",
  },
  {
    "name": "第 70 章",
    "url": "https://www.trongyi.vip/109_109270/70.html",
  },
  {
    "name": "第 71 章",
    "url": "https://www.trongyi.vip/109_109270/71.html",
  },
  {
    "name": "第 72 章",
    "url": "https://www.trongyi.vip/109_109270/72.html",
  },
  {
    "name": "第 73 章",
    "url": "https://www.trongyi.vip/109_109270/73.html",
  },
  {
    "name": "第 74 章",
    "url": "https://www.trongyi.vip/109_109270/74.html",
  },
  {
    "name": "第 75 章",
    "url": "https://www.trongyi.vip/109_109270/75.html",
  },
  {
    "name": "第 76 章",
    "url": "https://www.trongyi.vip/109_109270/76.html",
  },
  {
    "name": "第 77 章",
    "url": "https://www.trongyi.vip/109_109270/77.html",
  },
  {
    "name": "第 78 章",
    "url": "https://www.trongyi.vip/109_109270/78.html",
  },
  {
    "name": "第 79 章",
    "url": "https://www.trongyi.vip/109_109270/79.html",
  },
  {
    "name": "第 80 章",
    "url": "https://www.trongyi.vip/109_109270/80.html",
  },
  {
    "name": "第 81 章",
    "url": "https://www.trongyi.vip/109_109270/81.html",
  },
  {
    "name": "第 82 章",
    "url": "https://www.trongyi.vip/109_109270/82.html",
  },
  {
    "name": "第 83 章",
    "url": "https://www.trongyi.vip/109_109270/83.html",
  },
  {
    "name": "第 84 章",
    "url": "https://www.trongyi.vip/109_109270/84.html",
  },
  {
    "name": "第 85 章",
    "url": "https://www.trongyi.vip/109_109270/85.html",
  },
  {
    "name": "第 86 章",
    "url": "https://www.trongyi.vip/109_109270/86.html",
  },
  {
    "name": "第 87 章",
    "url": "https://www.trongyi.vip/109_109270/87.html",
  },
  {
    "name": "第 88 章",
    "url": "https://www.trongyi.vip/109_109270/88.html",
  },
  {
    "name": "第 89 章",
    "url": "https://www.trongyi.vip/109_109270/89.html",
  },
  {
    "name": "第 90 章",
    "url": "https://www.trongyi.vip/109_109270/90.html",
  },
  {
    "name": "第 91 章",
    "url": "https://www.trongyi.vip/109_109270/91.html",
  },
  {
    "name": "第 92 章",
    "url": "https://www.trongyi.vip/109_109270/92.html",
  },
  {
    "name": "第 93 章",
    "url": "https://www.trongyi.vip/109_109270/93.html",
  },
  {
    "name": "第 94 章",
    "url": "https://www.trongyi.vip/109_109270/94.html",
  },
  {
    "name": "第 95 章",
    "url": "https://www.trongyi.vip/109_109270/95.html",
  },
  {
    "name": "第 96 章",
    "url": "https://www.trongyi.vip/109_109270/96.html",
  },
  {
    "name": "第 97 章",
    "url": "https://www.trongyi.vip/109_109270/97.html",
  },
  {
    "name": "第 98 章",
    "url": "https://www.trongyi.vip/109_109270/98.html",
  },
  {
    "name": "第 99 章",
    "url": "https://www.trongyi.vip/109_109270/99.html",
  },
  {
    "name": "第 100 章",
    "url": "https://www.trongyi.vip/109_109270/100.html",
  },
  {
    "name": "第 101 章",
    "url": "https://www.trongyi.vip/109_109270/101.html",
  },
  {
    "name": "第 102 章",
    "url": "https://www.trongyi.vip/109_109270/102.html",
  },
  {
    "name": "第 103 章",
    "url": "https://www.trongyi.vip/109_109270/103.html",
  },
  {
    "name": "第 104 章",
    "url": "https://www.trongyi.vip/109_109270/104.html",
  },
  {
    "name": "第 105 章",
    "url": "https://www.trongyi.vip/109_109270/105.html",
  },
  {
    "name": "第 106 章",
    "url": "https://www.trongyi.vip/109_109270/106.html",
  },
  {
    "name": "第 107 章",
    "url": "https://www.trongyi.vip/109_109270/107.html",
  },
  {
    "name": "第 108 章",
    "url": "https://www.trongyi.vip/109_109270/108.html",
  },
  {
    "name": "第 109 章",
    "url": "https://www.trongyi.vip/109_109270/109.html",
  },
  {
    "name": "第 110 章",
    "url": "https://www.trongyi.vip/109_109270/110.html",
  },
  {
    "name": "第 111 章",
    "url": "https://www.trongyi.vip/109_109270/111.html",
  },
  {
    "name": "第 112 章",
    "url": "https://www.trongyi.vip/109_109270/112.html",
  },
  {
    "name": "第 113 章",
    "url": "https://www.trongyi.vip/109_109270/113.html",
  },
  {
    "name": "第 114 章",
    "url": "https://www.trongyi.vip/109_109270/114.html",
  },
  {
    "name": "第 115 章",
    "url": "https://www.trongyi.vip/109_109270/115.html",
  },
  {
    "name": "第 116 章",
    "url": "https://www.trongyi.vip/109_109270/116.html",
  },
  {
    "name": "第 117 章",
    "url": "https://www.trongyi.vip/109_109270/117.html",
  },
  {
    "name": "第 118 章",
    "url": "https://www.trongyi.vip/109_109270/118.html",
  },
  {
    "name": "第 119 章",
    "url": "https://www.trongyi.vip/109_109270/119.html",
  },
  {
    "name": "第 120 章",
    "url": "https://www.trongyi.vip/109_109270/120.html",
  },
  {
    "name": "第 121 章",
    "url": "https://www.trongyi.vip/109_109270/121.html",
  },
  {
    "name": "第 122 章",
    "url": "https://www.trongyi.vip/109_109270/122.html",
  },
  {
    "name": "第 123 章",
    "url": "https://www.trongyi.vip/109_109270/123.html",
  },
  {
    "name": "第 124 章",
    "url": "https://www.trongyi.vip/109_109270/124.html",
  },
  {
    "name": "第 125 章",
    "url": "https://www.trongyi.vip/109_109270/125.html",
  },
  {
    "name": "第 126 章",
    "url": "https://www.trongyi.vip/109_109270/126.html",
  },
  {
    "name": "第 127 章",
    "url": "https://www.trongyi.vip/109_109270/127.html",
  },
  {
    "name": "第 128 章",
    "url": "https://www.trongyi.vip/109_109270/128.html",
  },
  {
    "name": "第 129 章",
    "url": "https://www.trongyi.vip/109_109270/129.html",
  },
  {
    "name": "第 130 章",
    "url": "https://www.trongyi.vip/109_109270/130.html",
  },
  {
    "name": "第 131 章",
    "url": "https://www.trongyi.vip/109_109270/131.html",
  },
  {
    "name": "第 132 章",
    "url": "https://www.trongyi.vip/109_109270/132.html",
  },
  {
    "name": "第 133 章",
    "url": "https://www.trongyi.vip/109_109270/133.html",
  },
  {
    "name": "第 134 章",
    "url": "https://www.trongyi.vip/109_109270/134.html",
  },
  {
    "name": "第 135 章",
    "url": "https://www.trongyi.vip/109_109270/135.html",
  },
  {
    "name": "第 136 章",
    "url": "https://www.trongyi.vip/109_109270/136.html",
  },
  {
    "name": "第 137 章",
    "url": "https://www.trongyi.vip/109_109270/137.html",
  },
  {
    "name": "第 138 章",
    "url": "https://www.trongyi.vip/109_109270/138.html",
  },
  {
    "name": "第 139 章",
    "url": "https://www.trongyi.vip/109_109270/139.html",
  },
  {
    "name": "第 140 章",
    "url": "https://www.trongyi.vip/109_109270/140.html",
  },
  {
    "name": "第 141 章",
    "url": "https://www.trongyi.vip/109_109270/141.html",
  },
  {
    "name": "第 142 章",
    "url": "https://www.trongyi.vip/109_109270/142.html",
  },
  {
    "name": "第 143 章",
    "url": "https://www.trongyi.vip/109_109270/143.html",
  },
  {
    "name": "第 144 章",
    "url": "https://www.trongyi.vip/109_109270/144.html",
  },
  {
    "name": "第 145 章",
    "url": "https://www.trongyi.vip/109_109270/145.html",
  },
  {
    "name": "第 146 章",
    "url": "https://www.trongyi.vip/109_109270/146.html",
  },
  {
    "name": "第 147 章",
    "url": "https://www.trongyi.vip/109_109270/147.html",
  },
  {
    "name": "第 148 章",
    "url": "https://www.trongyi.vip/109_109270/148.html",
  },
  {
    "name": "第 149 章",
    "url": "https://www.trongyi.vip/109_109270/149.html",
  },
  {
    "name": "第 150 章",
    "url": "https://www.trongyi.vip/109_109270/150.html",
  },
  {
    "name": "第 151 章",
    "url": "https://www.trongyi.vip/109_109270/151.html",
  },
  {
    "name": "第 152 章",
    "url": "https://www.trongyi.vip/109_109270/152.html",
  },
  {
    "name": "第 153 章",
    "url": "https://www.trongyi.vip/109_109270/153.html",
  },
  {
    "name": "第 154 章",
    "url": "https://www.trongyi.vip/109_109270/154.html",
  },
  {
    "name": "第 155 章",
    "url": "https://www.trongyi.vip/109_109270/155.html",
  },
  {
    "name": "第 156 章",
    "url": "https://www.trongyi.vip/109_109270/156.html",
  },
  {
    "name": "第 157 章",
    "url": "https://www.trongyi.vip/109_109270/157.html",
  },
  {
    "name": "第 158 章",
    "url": "https://www.trongyi.vip/109_109270/158.html",
  },
  {
    "name": "第 159 章",
    "url": "https://www.trongyi.vip/109_109270/159.html",
  },
  {
    "name": "第 160 章",
    "url": "https://www.trongyi.vip/109_109270/160.html",
  },
  {
    "name": "第 161 章",
    "url": "https://www.trongyi.vip/109_109270/161.html",
  },
  {
    "name": "第 162 章",
    "url": "https://www.trongyi.vip/109_109270/162.html",
  },
  {
    "name": "第 163 章",
    "url": "https://www.trongyi.vip/109_109270/163.html",
  },
  {
    "name": "第 164 章",
    "url": "https://www.trongyi.vip/109_109270/164.html",
  },
  {
    "name": "第 165 章",
    "url": "https://www.trongyi.vip/109_109270/165.html",
  },
  {
    "name": "第 166 章",
    "url": "https://www.trongyi.vip/109_109270/166.html",
  },
  {
    "name": "第 167 章",
    "url": "https://www.trongyi.vip/109_109270/167.html",
  },
  {
    "name": "第 168 章",
    "url": "https://www.trongyi.vip/109_109270/168.html",
  },
  {
    "name": "第 169 章",
    "url": "https://www.trongyi.vip/109_109270/169.html",
  },
  {
    "name": "第 170 章",
    "url": "https://www.trongyi.vip/109_109270/170.html",
  },
  {
    "name": "第 171 章",
    "url": "https://www.trongyi.vip/109_109270/171.html",
  },
  {
    "name": "第 172 章",
    "url": "https://www.trongyi.vip/109_109270/172.html",
  },
  {
    "name": "第 173 章",
    "url": "https://www.trongyi.vip/109_109270/173.html",
  },
  {
    "name": "第 174 章",
    "url": "https://www.trongyi.vip/109_109270/174.html",
  },
  {
    "name": "第 175 章",
    "url": "https://www.trongyi.vip/109_109270/175.html",
  },
  {
    "name": "第 176 章",
    "url": "https://www.trongyi.vip/109_109270/176.html",
  },
  {
    "name": "第 177 章",
    "url": "https://www.trongyi.vip/109_109270/177.html",
  },
  {
    "name": "第 178 章",
    "url": "https://www.trongyi.vip/109_109270/178.html",
  },
  {
    "name": "第 179 章",
    "url": "https://www.trongyi.vip/109_109270/179.html",
  },
  {
    "name": "第 180 章",
    "url": "https://www.trongyi.vip/109_109270/180.html",
  },
  {
    "name": "第 181 章",
    "url": "https://www.trongyi.vip/109_109270/181.html",
  },
  {
    "name": "第 182 章",
    "url": "https://www.trongyi.vip/109_109270/182.html",
  },
  {
    "name": "第 183 章",
    "url": "https://www.trongyi.vip/109_109270/183.html",
  },
  {
    "name": "第 184 章",
    "url": "https://www.trongyi.vip/109_109270/184.html",
  },
  {
    "name": "第 185 章",
    "url": "https://www.trongyi.vip/109_109270/185.html",
  },
  {
    "name": "第 186 章",
    "url": "https://www.trongyi.vip/109_109270/186.html",
  },
  {
    "name": "第 187 章",
    "url": "https://www.trongyi.vip/109_109270/187.html",
  },
  {
    "name": "第 188 章",
    "url": "https://www.trongyi.vip/109_109270/188.html",
  },
  {
    "name": "第 189 章",
    "url": "https://www.trongyi.vip/109_109270/189.html",
  },
  {
    "name": "第 190 章",
    "url": "https://www.trongyi.vip/109_109270/190.html",
  },
  {
    "name": "第 191 章",
    "url": "https://www.trongyi.vip/109_109270/191.html",
  },
  {
    "name": "第 192 章",
    "url": "https://www.trongyi.vip/109_109270/192.html",
  },
  {
    "name": "第 193 章",
    "url": "https://www.trongyi.vip/109_109270/193.html",
  },
  {
    "name": "第 194 章",
    "url": "https://www.trongyi.vip/109_109270/194.html",
  },
  {
    "name": "第 195 章",
    "url": "https://www.trongyi.vip/109_109270/195.html",
  },
  {
    "name": "第 196 章",
    "url": "https://www.trongyi.vip/109_109270/196.html",
  },
  {
    "name": "第 197 章",
    "url": "https://www.trongyi.vip/109_109270/197.html",
  },
  {
    "name": "第 198 章",
    "url": "https://www.trongyi.vip/109_109270/198.html",
  },
  {
    "name": "第 199 章",
    "url": "https://www.trongyi.vip/109_109270/199.html",
  },
  {
    "name": "第 200 章",
    "url": "https://www.trongyi.vip/109_109270/200.html",
  },
  {
    "name": "第 201 章",
    "url": "https://www.trongyi.vip/109_109270/201.html",
  },
  {
    "name": "第 202 章",
    "url": "https://www.trongyi.vip/109_109270/202.html",
  },
  {
    "name": "第 203 章",
    "url": "https://www.trongyi.vip/109_109270/203.html",
  },
  {
    "name": "第 204 章",
    "url": "https://www.trongyi.vip/109_109270/204.html",
  },
  {
    "name": "第 205 章",
    "url": "https://www.trongyi.vip/109_109270/205.html",
  },
  {
    "name": "第 206 章",
    "url": "https://www.trongyi.vip/109_109270/206.html",
  },
  {
    "name": "第 207 章",
    "url": "https://www.trongyi.vip/109_109270/207.html",
  },
  {
    "name": "第 208 章",
    "url": "https://www.trongyi.vip/109_109270/208.html",
  },
  {
    "name": "第 209 章",
    "url": "https://www.trongyi.vip/109_109270/209.html",
  },
  {
    "name": "第 210 章",
    "url": "https://www.trongyi.vip/109_109270/210.html",
  },
  {
    "name": "第 211 章",
    "url": "https://www.trongyi.vip/109_109270/211.html",
  },
]
`;
